packages/prover/src/interfaces.ts

```diff
@@ -78,6 +79,10 @@ export type ConsensusNodeOptions =
   | {transport: LCTransport.Rest; urls: string[]}
   | {transport: LCTransport.P2P; bootnodes: string[]};
 
-export type VerifiedExecutionInitOptions = LogOptions &
-  ConsensusNodeOptions &
-  NetworkOrConfig & {wsCheckpoint?: string; signal?: AbortSignal};
+export type RootProviderOptions = {
+  wsCheckpoint?: string;
+  signal?: AbortSignal;
+  unverifiedWhitelist?: string[];
+};
```

Comment:
This is really nuanced but sorting options based on actual options (e.g. wsCheckpoint) and modules (e.g. signal) might be helpful

```suggestion
export type RootProviderOptions = {
  signal?: AbortSignal;
  wsCheckpoint?: string;
  unverifiedWhitelist?: string[];
};
```